### PR TITLE
Bugs and fixes

### PR DESCRIPTION
**Pull request by dmitrykuzmenkov**

### CODE REVIEW

File: src/lib/FileStorage.php [hunk 1]
```diff
@@ -337,7 +348,7 @@ public function getBackupPaths(): array {
       }
 
       // Backup directory consists of next folders
-      // data - indexes stored here (from data dir and other files from there)
+      // data - tabless stored here (from data dir and other files from there)
```
Comment (sanikolaev) [inline src/lib/FileStorage.php:351]:
```suggestion
      // data - tables stored here (from data dir and other files from there)
```

File: src/lib/ManticoreBackup.php [hunk 2]
```diff
@@ -32,31 +33,60 @@ public static function store(ManticoreClient $Client, FileStorage $Storage, arra
...
+      println(LogLevel::Info, 'Backing up global state files...');
+      $files = $Client->getConfig()->getStatePaths();
+      $is_ok = $Storage->copyPaths($files, $destination['state'], true);
+      println(LogLevel::Info, '  global state files – ' . get_op_result($is_ok));
+
+      // - backup external files for each index
+      println(LogLevel::Info, 'Backing up external index files...');
+      foreach ($tables as $index => $type) {
+        println(LogLevel::Info, '  ' . $index . ' (' . $type . ')...');
+        // Show settings on distributed tables returns error 500 so we simply skip it
```
Comment (sanikolaev) [inline src/lib/ManticoreBackup.php:61]:
```suggestion
        // SHOW SETTINGS is not supported for distributed tables, so we simply skip it
```

File: src/lib/ManticoreBackup.php [hunk 3]
```diff
@@ -32,31 +33,60 @@ public static function store(ManticoreClient $Client, FileStorage $Storage, arra
...
+        if ($type === 'distributed') {
+          println(LogLevel::Info, '   ⏩ SKIP');
```
Comment (sanikolaev) [inline src/lib/ManticoreBackup.php:63]: Didn't we want to remove emojis so the output looks nice in `less` etc?

File: test/ManticoreBackupTest.php [hunk 4]
```diff
@@ -257,7 +271,7 @@ class ManticoreMockedClient extends ManticoreClient {
   protected Closure $timeout_fn;
 
   /**
-   * Set delay timeout that we will use in case of calling freeze indexes
+   * Set delay timeout that we will use in case of calling freeze tables
```
Comment (sanikolaev) [inline test/ManticoreBackupTest.php:274]: Pls remind me why we need the timeout.

**Comment by sanikolaev** [comment]
# Notes after functional testing

## Copyright

```
[root@0391e5b35b86 ~]# /manticoresearch-backup/build/manticore_backup
2022-10-03 15:34:24 [Error] Failed to find target dir to store backup: none
```

Let's show copyright here too similar to `searchd`/`indexer`/`indextool`:

```
[root@0391e5b35b86 ~]# searchd -ksjjfkd
Manticore 5.0.3 79a994220@220924 dev (columnar 1.16.1 abd8d4d@220927)
Copyright (c) 2001-2016, Andrew Aksyonoff
Copyright (c) 2008-2016, Sphinx Technologies Inc (http://sphinxsearch.com)
Copyright (c) 2017-2022, Manticore Software LTD (https://manticoresearch.com)

[35:04.073] [1821] FATAL: malformed or unknown option near '-ksjjfkd'; use '-h' or '--help' to see available options.
```

It should look like this:

```
[root@0391e5b35b86 ~]# /manticoresearch-backup/build/manticore_backup
Copyright (c) 2017-2022, Manticore Software LTD (https://manticoresearch.com)
Copyright (c) 2001-2016, Andrew Aksyonoff
Copyright (c) 2008-2016, Sphinx Technologies Inc (http://sphinxsearch.com)

2022-10-03 15:34:24 [Error] Failed to find target dir to store backup: none
```

# Emojis

Looks not good in `less`:

image

I'm sure there's a way to have emojis displayed ok in `less`, `vim` etc, but I'm not sure it's easy for most, so I suggest we don't use emojis in output of `manticore_backup`.

File: src/lib/ManticoreBackup.php (hunk 3 above)
Comment (dmitrykuzmenkov) [reply src/lib/ManticoreBackup.php:63]: It seems like this is the only one emoji that I've missed, will fix it

File: test/ManticoreBackupTest.php (hunk 4 above)
Comment (dmitrykuzmenkov) [reply test/ManticoreBackupTest.php:274]: This is mocked class used only for testing purpose, it's required to do timeout to emulate test cases with interruption of the process in the middle. It's just workaround for tests only

**Review by sanikolaev — APPROVED** [review]

**Pull request merged**